pkg/kapp/resources/mod_path.go

```diff
@@ -21,6 +21,7 @@ type Path []*PathPart
 
 type PathPart struct {
 	MapKey     *string
+	RegexObj   *PathPartRegex
```

Comment:
Since it's already an object, should it just be `Regex` and not `RegexObj`? (Then we will have to use `Regex.Regex` in some places which might sound weird but I couldn't come up with any other alternative)

Reply:
Adopted